pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,463 @@
+                                reportLinks.push(report[1].URL.v.v);
+                            }
+                            reported = true;
+                        }
+                    }
+
+                    this.setState({
+                        problemState: reported ? ProblemState.REPORTED : this.state.problemState,
+                        reportLinks: reportLinks,
+                    });
+                });
+        get_problem_property(this.props.problem, "CanBeReported")
+                .promise()
+                .then((reportable) => {
+                    if (!reportable) {
+                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                    }
+                });
+    }
+
+    onCancelButtonClick(event) {
+        this.state.process.close("canceled");
+    }
+
+    onReportButtonClick(event) {
+        this.setState({ problemState: ProblemState.UNREPORTABLE });
+
+        const process = cockpit.spawn(["reporter-ureport", "-d", this.props.problem.ID],
+                                      {
+                                          err: "out",
+                                          superuser: "true",
+                                      })
+                .always(() => this.updateStatusFromBus())
+                .done(() => this.setState({ problemState: ProblemState.REPORTED, }))
+                .fail((ex) => {
+                    this.setState({ problemState: ProblemState.REPORTABLE, });
+
+                    if (ex.exit_signal != null) {
+                        console.error(cockpit.format("reporter-ureport was killed with signal $0", ex.exit_signal));
+                    }
+                })
+                .stream((data) => this.setState({ message: data, }));
+
+        this.setState({
+            problemState: ProblemState.REPORTING,
+            process: process,
+        });
+    }
+
+    render() {
+        return <WorkflowRow label={_("Report to ABRT Analytics")}
+                            message={this.state.message}
+                            onCancelButtonClick={this.onCancelButtonClick}
+                            onReportButtonClick={this.onReportButtonClick}
+                            problemState={this.state.problemState}
+                            reportLinks={this.state.reportLinks} />;
+    }
+}
+
+const reportd_client = cockpit.defer();
+const permission = cockpit.permission({ admin: true });
+permission.addEventListener("changed", () => {
+    // reportd may use ABRT API that requires authorization, but it cannot
+    // be given using polkit, as the calling process will always be reportd
+    // and not cockpit-bridge. However, UID 0 is always authorized, hence
+    // the spawning of the system service and using the system bus here.
+    //
+    // TODO: Only use system bus when reportd is merged into ABRT
+    //       (https://github.com/abrt/reportd/issues/8).
+    const bus_type = permission.allowed ? "system" : "session";
```

Comment:
That PR didn't land yet, does that block landing this PR? Does it work to call it on the session bus as non-root, and does it work to call it on the system bus without that PR?

Reply:
Actually, the work on that is sort of blocked by this, since I can iterate faster with reportd being a separate project, but since I more or less imagine what the requirements are, I can go ahead with that soon™.

It does work if a non-root user speaks to it over the session bus, but the main point of the comment is that the ABRT daemon is only reachable via system bus.

Reply:
Ah, I see! Sorry, I previously misunderstood the comment before.